### ISSUE TITLE
Livepush fails to push changes with errors `Cannot remove '/tmp/balena': Device or resource busy`

### ISSUE THREAD

**Issue by xginn8** [balena-io/balena-cli#1437]
- **balena CLI version:** 11.10.0
- **Operating system version:** arch linux (64 bit)

---

When pushing to BoB via livepush, I often have builds crash with the following errors in my local log:

```
[Live]    [devices]    rm: cannot remove '/tmp/balena': Device or resource busy
[Live]    [devices]    rm: cannot remove '/tmp/resin': Device or resource busy
[Error]   [devices]   Command JOBS=MAX npm install --unsafe-perm --production   && npm cache clean --force 2>/dev/null  && rm -rf /tmp/* failed with exit code: 1
```

Just touching the Dockerfile starts a rebuild and that's enough, and I realize cache cleaning is probably unwise when building incrementally locally. These errors are _only_ seen on subsequent builds, ie when the device is in steady state (long after startup).

**Comment by pdcastro** [comment]
CC: @CameronDiver

**Comment by pdcastro** [comment]
@xginn8, to confirm, is `rm -rf /tmp/*` a command in the Dockerfile?

> [Live]    [devices]    rm: cannot remove '/tmp/balena': Device or resource busy

Is that `/tmp/balena` folder on the device? It happens to be a folder used by the supervisor for application update locks: https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/

I am not saying it's the cause of the problem, it's just something that occurred to me.

**Comment by xginn8** [comment]
> @xginn8, to confirm, is `rm -rf /tmp/*` a command in the Dockerfile?
That is correct, specifically used here: https://github.com/balena-io/resin-proxy/blob/master/Dockerfile#L17. I believe this is a common way to reduce the final size of images, though obviously as noted removing the cache makes subsequent building much less performant.
> 
> > [Live]    [devices]    rm: cannot remove '/tmp/balena': Device or resource busy
> 
> Is that `/tmp/balena` folder on the device? It happens to be a folder used by the supervisor for application update locks: https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/
> 
> I am not saying it's the cause of the problem, it's just something that occurred to me.
That makes lots of sense to me!
